active/teachingtip/TeachingTipSpec.md

```diff
@@ -0,0 +1,690 @@
+                args.Cancel = true;
+                ShowLastErrorMessage();
+            }
+        }
+    }
+}
+```
+
+# Remarks
+
+### Related articles 
+
+* [Dialogs and flyouts](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/dialogs-and-flyouts/index)
+
+### Recommendations
+* Tips are impermanent and should not contain information or options that are critical to the experience of an application. 
+* Try to avoid showing teaching tips too often. Teaching tips are most likely to each recieve individual attention when they are staggered throughout long sessions or across multiple sessions.    
+* Keep tips succinct and their topic clear. Research shows users, on average, only read 3-5 words and only comprehend 2-3 words before deciding whether to interact with a tip. 
+
+# API Notes
+
+### Notable Properties  
+
+| Name | Description |
+|:-:|:--|
+| PointerMode | Gets or sets a value that indicates whether to show the pointer for a teaching tip. |
+| PreferredPlacement  | Gets or sets the default placement to be used for the teaching tip. |
+| ShouldConstrainToRootBounds | Gets or sets a value that indicates whether the teaching tip will constrain to the bounds of its root. |
+
+### Events    
+
+| Name | Description |
+|:-:|:--|
+| ActionButtonClick | Occurs after the action button has been tapped. |
+| CloseButtonClick | Occurs after the close button has been tapped. |
+| Closed | Occurs after the tip is closed. |
+| Closing |Occurs just before the tip begins to close. |
+
+# API Details 
+
+```c++ 
+enum TeachingTipCloseReason
+{
+    CloseButton,
+    LightDismiss,
+    Programmatic,
+};
+
+enum TeachingTipPlacementMode
+{
+    Auto,
+    Top,
+    Bottom,
+    Left,
+    Right,
+    TopEdgeAlignedRight,
+    TopEdgeAlignedLeft,
+    BottomEdgeAlignedRight,
+    BottomEdgeAlignedLeft,
+    LeftEdgeAlignedTop,
+    LeftEdgeAlignedBottom,
+    RightEdgeAlignedTop,
+    RightEdgeAlignedBottom,
+};
+
+enum TeachingTipHeroContentPlacementMode
+{
+    Auto,
+    Top,
+    Bottom,
+};
+
+enum TeachingTipPointerMode
+{
+	Auto,
+	On,
+	Off,
+};
+
+runtimeclass TeachingTipClosedEventArgs
+{
+    TeachingTipCloseReason Reason{ get; };
+};
+
+runtimeclass TeachingTipClosingEventArgs
+{
+    TeachingTipCloseReason Reason{ get; };
+    Boolean Cancel;
+    Windows.Foundation.Deferral GetDeferral();
+};
+
+unsealed runtimeclass TeachingTipTemplateSettings : Windows.UI.Xaml.DependencyObject
+{
+    TeachingTipTemplateSettings();
+
+    Windows.UI.Xaml.Thickness TopRightHighlightMargin;
+    Windows.UI.Xaml.Thickness TopLeftHighlightMargin;
+
+    Windows.UI.Xaml.Controls.IconElement IconElement;
+
+    static Windows.UI.Xaml.DependencyProperty TopRightHighlightMarginProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty TopLeftHighlightMarginProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty IconElementProperty{ get; };
+}
+
+[MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
```

Comment:
We shouldn't have any of these [MUX] attributes.

Reply:
Removed.